Google.PowerShell.IntegrationTests/Container/Container.GkeCluster.Tests.ps1

```diff
@@ -345,14 +348,118 @@ Describe "Add-GkeCluster" {
+    $clusterTwoZone = "us-west1-b"
+
+    $clusterThreeName = "gcp-new-gkecluster-3-$r"
+    $clusterThreeZone = "europe-west1-c"
+
+    $clusterFourName = "gcp-new-gkecluster-4-$r"
+    $clusterFourZone = "asia-east1-a"
+
+    $clusterOneParameter = @{"clusterName" = $clusterOneName}
+    $clusterTwoParameter = @{"clusterName" = $clusterTwoName; "zone" = $clusterTwoZone}
+    $clusterThreeParameter = @{"clusterName" = $clusterThreeName; "zone" = $clusterThreeZone}
+    $clusterFourParameter = @{"clusterName" = $clusterFourName; "zone" = $clusterFourZone}
+
+    # Start creating the cluster.
+    $clusterOneJob = Start-Job -ScriptBlock $clusterCreationWithoutUsingNodeConfigScriptBlock `
+                               -ArgumentList @($gcloudCmdletsPath, $clusterOneParameter)
+    $clusterTwoJob = Start-Job -ScriptBlock $clusterCreationWithoutUsingNodeConfigScriptBlock `
+                               -ArgumentList @($gcloudCmdletsPath, $clusterTwoParameter)
+    $clusterThreeJob = Start-Job -ScriptBlock $clusterCreationWithoutUsingNodeConfigScriptBlock `
+                               -ArgumentList @($gcloudCmdletsPath, $clusterThreeParameter)
+    $clusterFourJob = Start-Job -ScriptBlock $clusterCreationWithoutUsingNodeConfigScriptBlock `
+                               -ArgumentList @($gcloudCmdletsPath, $clusterFourParameter)
+
+    Wait-Job $clusterOneJob, $clusterTwoJob, $clusterThreeJob, $clusterFourJob | Remove-Job
+
+    # Check that all the clusters are really created.
+    Check-Cluster $clusterOneName $zone
+    Check-Cluster $clusterTwoName $clusterTwoZone
+    Check-Cluster $clusterThreeName $clusterThreeZone
+    Check-Cluster $clusterFourName $clusterFourZone
+
+    # Running jobs to remove these 3 clusters in parallel to minimize test run time.
+    $clusterTwoRemovalJob = Start-Job -ScriptBlock {
+        param($cmdletPath, $clusterName, $clusterZone)
+        # For this cluster, we remove by pipeline.
+        . $cmdletPath
+        Install-GCloudCmdlets | Out-Null
+        Get-GkeCluster -ClusterName $clusterName -Zone $clusterZone | Remove-GkeCluster
+    } -ArgumentList @($gcloudCmdletsPath, $clusterTwoName, $clusterTwoZone)
+
+    $clusterThreeRemovalJob = Start-Job -ScriptBlock {
+        param($cmdletPath, $clusterName, $clusterZone)
+        # For this cluster, we remove by object.
+        . $cmdletPath
+        Install-GCloudCmdlets | Out-Null
+        $cluster = Get-GkeCluster -ClusterName $clusterName -Zone $clusterZone
+        Remove-GkeCluster $cluster
+    } -ArgumentList @($gcloudCmdletsPath, $clusterThreeName, $clusterThreeZone)
+
+    $clusterFourRemovalJob = Start-Job -ScriptBlock {
+        param($cmdletPath, $clusterName, $clusterZone)
+        # For this cluster, we used -Zone in the removal.
+        . $cmdletPath
+        Install-GCloudCmdlets | Out-Null
+        Remove-GkeCluster $clusterName -Zone $clusterZone
+    } -ArgumentList @($gcloudCmdletsPath, $clusterFourName, $clusterFourZone)
```

Comment:
I feel like these jobs should be named after what they are doing (by pipeline etc.) rather than the cluster number they are doing it to. I read "should remove cluster by pipeline", then had to scroll up check that `$clusterTwoRemovalJob` did in fact remove using the pipeline.